### ISSUE TITLE
Are you serious?

### ISSUE THREAD

**Issue by stan-ko** [rampo/UpdateChecker#123]
You keep an Activity in a static reference in UpdateChecker class - are you an Android developer really?

**Comment by AdityaAnand1** [comment]
Agreed. Everything is needlessly static. Although, there's nothing wrong with that but if you're going to go down that road then a public static init() method is more appropriate than using a constructor. Not to mention the static activity reference is passed on to an AsyncTask without any attempts at cleaning up which could lead to memory leaks.
